src/View/mainpage/DrawROIWindow/UIDrawROIWindow.py

```diff
@@ -862,29 +880,146 @@ def onFillClicked(self):
-                    self.keep_empty_pixel,
-                    self.pixel_transparency,
-                    set()
-                )
-                self.slice_changed = True
-                self.has_drawing = True
-                self.dicom_view.view.setScene(self.drawingROI)
-                self.enable_cursor_diameter_change_box()
+                pixmaps = self.patient_dict_container.get("pixmaps_axial")
+                id = self.current_slice
+
+                # This needs to be updated when merged with 2d
+                print("is_3d: " + str(is_3d))
+                if is_3d:
+                    print(str(is_3d) + ": is_3d")
+                    if hasattr(self, 'seed'):
+                        delattr(self, 'seed')
+                    self.create_drawing_3D(min_pixel, max_pixel, pixmaps, id)
+                else:
+                    self.create_drawing(min_pixel, max_pixel, pixmaps, id)
+
             else:
                 QMessageBox.about(self.draw_roi_window_instance,
                                   "Not Enough Data",
                                   "Not all values are specified or correct.")
-        logging.debug("onFillClicked finished")
+
+    def create_drawing(self, min_pixel, max_pixel, pixmaps, id):
+        """Creates 3d drawing using BFS"""
+        dt = self.patient_dict_container.dataset[id]
+        dt.convert_pixel_data()
+        # Path to the selected .dcm file
+        location = self.patient_dict_container.filepaths[id]
+        self.ds = pydicom.dcmread(location)
+
+        self.drawingROI = Drawing(
+            pixmaps[id],
+            dt._pixel_array.transpose(),
+            min_pixel,
+            max_pixel,
+            self.patient_dict_container.dataset[id],
+            self.draw_roi_window_instance,
+            self.slice_changed,
+            self.current_slice,
+            self.drawing_tool_radius,
+            self.keep_empty_pixel,
+            self.pixel_transparency,
+            set()
+        )
+
+        self.slice_changed = True
+        self.has_drawing = True
+        self.dicom_view.view.setScene(self.drawingROI)
+        self.enable_cursor_diameter_change_box()
+
+    def create_drawing_3D(self, min_pixel, max_pixel, pixmaps, id):
+        """
+        Creates drawing across multiple slides allowing for the user to start drawing on dicom view.
+        """
+        # If the seed is set then start searching, else assign the drawing function to the left click
+        if hasattr(self, 'seed'):
+            # Search down from the start position then up from the start position
+            for x in range(0, 2):
+                print("x == " + str(x))
+                if x == 0:
+                    range_start = id - 1
+                    range_end = 1
+                    step = -1
+                else:
+                    range_start = id
+                    range_end = len(self.patient_dict_container.dataset) - 1
+                    step = 1
```

Comment:
Im assuming this part is so that it doesn't have to be that specific pixel? If so you could probably better describe that maybe in docstring or somthing

Reply:
actually now im still not sure haha. Is it the slices above and below?